### DISCUSSION TITLE
What is the expected behaviour of setting a valueAsNumber field to null?

### DISCUSSION THREAD

**Discussion by jameshoward** [react-hook-form/react-hook-form#8953] [Q&A]
If we have a form field registered with `valueAsNumber` set to `true` and then call `setValue('fieldName', null)` the value of the form field (e.g. from a `useWatch`) becomes `NaN`.

Is `null` not a valid value for a numeric field?

Interestingly setting it to `undefined` does maintain the value as `undefined`, but I believe I read somewhere in the docs that we shouldn't try to set values to `undefined`.

This changed in [8487](https://github.com/react-hook-form/react-hook-form/pull/8487) as it used to change the value to `0`, which would also submit `0` for the field.

CodeSandbox example (using v7.34.2): https://codesandbox.io/s/react-hook-form-basic-forked-0l26ro?file=/src/index.js

There is a workaround if you do want numeric fields to be allowed to be `null`. I'm using `setValueAs` instead of `valueAsNumber` and using something like:

```
(value: any): number | null | undefined => {
  if (value === null || value === undefined) {
    return value;
  }

  if (value === '') {
    return NaN;
  }

  return +value;
};
```

**Comment by bluebill1049** [comment]
use `setValueAs` for custom (your preferred use case and parsing result) as it's more flexible and less opinionated on the result return.
